src/java.base/share/classes/sun/security/ssl/SSLSessionContextImpl.java

```diff
@@ -69,6 +72,11 @@ final class SSLSessionContextImpl implements SSLSessionContext {
     private int cacheLimit;             // the max cache size
     private int timeout;                // timeout in seconds
 
+    private int currentKeyID = new SecureRandom().nextInt();
```

Comment:
Could this instantiation of new SecureRandom become a concurrency bottleneck on entropy? Should the SecureRandom from SSLContext be injected to constructor?

Reply:
Hm, I'm not sure if this results in a real problem but I agree that it is better to be on the safe side :)

From my understanding (and please correct me if I'm wrong) `SSLContextImpl::engineInit()` has to be called before we can start to negotiate a connection and create a session key (i.e. before `SSLContextImpl::getKey()` and subsequently `SSLSessionContextImpl::getCurrentKeyID()` will be called for the first time on a context).

This means that we can use the random "*priming date*" which is created in `SSLContextImpl::engineInit()` anyway (thanks for that link :) to initialize `SSLSessionContextImpl::currentKeyID` and don't need any additional call to `SecureRandom` at all.